sycl/plugins/unified_runtime/pi2ur.hpp

```diff
@@ -70,7 +81,7 @@ class ConvertHelper : public ReturnHelper {
 public:
   // Convert the value using a conversion map
   template <typename TypeUR, typename TypePI>
-  pi_result convert(const std::unordered_map<TypeUR, TypePI> &Map) {
+  pi_result convert(std::function<std::optional<TypePI>(TypeUR)> Func) {

```

Comment:
Why do you need the `optional` stuff at all? A conversion should always be defined, and we should `die` at line 99 if it is not

Reply:
I guess I see, you return std::nullopt when there is no conversion defined, and then die if it's conversion to bool is "false".  Why not just die directly in the conversion switches directly?

Reply:
I moved die as default in switch statement and remove std::optional